metrics/prometheus/prometheus.go

```diff
@@ -0,0 +1,300 @@
+	ExtraTTLSeconds prometheus.Histogram
+}
+
+func CreatePrometheusMetrics(cfg config.PrometheusMetrics) *PrometheusMetrics {
+	timeBuckets := []float64{0.001, 0.002, 0.005, 0.01, 0.025, 0.05, 0.1, 0.2, 0.3, 0.4, 0.5, 1}
+	requestSizeBuckets := []float64{0.01, 0.025, 0.05, 0.1, 0.2, 0.3, 0.4, 0.5, 1}
+	registry := prometheus.NewRegistry()
+	promMetrics := &PrometheusMetrics{
+		Registry: registry,
+		Puts: &PrometheusRequestStatusMetric{
+			Duration: newHistogram(cfg, registry,
+				"puts_request_duration",
+				"Duration in seconds Prebid Cache takes to process put requests.",
+				timeBuckets,
+			),
+			RequestStatus: newCounterVecWithLabels(cfg, registry,
+				"puts_request",
+				"Count of total requests to Prebid Server labeled by status.",
+				[]string{StatusKey},
+			),
+		},
+		Gets: &PrometheusRequestStatusMetric{
+			Duration: newHistogram(cfg, registry,
+				"gets_request_duration",
+				"Duration in seconds Prebid Cache takes to process get requests.",
+				timeBuckets,
+			),
+			RequestStatus: newCounterVecWithLabels(cfg, registry,
+				"gets_request",
+				"Count of total get requests to Prebid Server labeled by status.",
+				[]string{StatusKey},
+			),
+		},
+		PutsBackend: &PrometheusRequestStatusMetricByFormat{
+			Duration: newHistogram(cfg, registry,
+				"puts_backend_duration",
+				"Duration in seconds Prebid Cache takes to process backend put requests.",
+				timeBuckets,
+			),
+			PutBackendRequests: newCounterVecWithLabels(cfg, registry,
+				"puts_backend",
+				"Count of total requests to Prebid Cache labeled by format, status and whether or not it comes with TTL",
+				[]string{FormatKey},
+			),
+			RequestLength: newHistogram(cfg, registry,
+				"puts_backend_request_size_bytes",
+				"Size in bytes of a backend put request.",
+				requestSizeBuckets,
+			),
+		},
+		GetsBackend: &PrometheusRequestStatusMetric{
+			Duration: newHistogram(cfg, registry,
+				"gets_backend_duration",
+				"Duration in seconds Prebid Cache takes to process backend get requests.",
+				timeBuckets,
+			),
+			RequestStatus: newCounterVecWithLabels(cfg, registry,
+				"gets_backend",
+				"Count of total backend get requests to Prebid Server labeled by status.",
+				[]string{StatusKey},
+			),
+		},
+		Connections: &PrometheusConnectionMetrics{
+			ConnectionsClosed: newSingleCounter(cfg, registry, ConnOpenedKey, "Count the number of open connections"),
+			ConnectionsOpened: newSingleCounter(cfg, registry, ConnClosedKey, "Count the number of closed connections"),
+			ConnectionsErrors: newCounterVecWithLabels(cfg, registry,
+				ConnErrorKey,
+				"Count the number of connection accept errors or connection close errors",
+				[]string{ConnErrorKey},
+			),
+		},
+		ExtraTTL: &PrometheusExtraTTLMetrics{
+			ExtraTTLSeconds: newHistogram(cfg, registry,
+				"extra_ttl_seconds",
+				"Extra time to live in seconds specified",
+				timeBuckets,
+			),
+		},
+	}
+	promMetrics.ExtraTTL.ExtraTTLSeconds.Observe(5000.00)
```

Comment:
Was this for debugging purposes?

Reply:
No, it was there before this PR. Was there on the Influx side so I translated it to prometheus

Reply:
I don't understand it's purpose. Please consider removing it.

Reply:
Please remove this.